internal/command/command.go

```diff
@@ -355,10 +356,246 @@ var CmdUpdateApis = &Command{
+	libraries := pipelineState.LibraryReleaseStates
+	var prDescription string
+	var librariesToRelease map[string]string
+	librariesToRelease = make(map[string]string)
+	for i := 0; i < len(libraries); i++ {
+		library := libraries[i]
+		var commitMessages []string
+		//TODO: need to add common paths as well as refactor to see if can check all paths at 1 x
+		for j := 0; j < len(library.SourcePaths); j++ {
+			//TODO: figure out generic logic
+			previousReleaseTag := library.Id + "-" + library.CurrentVersion
+			commits, err := gitrepo.GetApiCommitsSinceTagForSource(repo, library.SourcePaths[j], previousReleaseTag)
+			if err != nil {
+				fmt.Println("Error searching commits:", err)
+				//TODO update PR description with this data and mark as not humanly resolvable
+			}
+			for _, commit := range commits {
+				commitMessages = append(commitMessages, commit.Message)
+			}
+		}
+
+		if len(commitMessages) > 0 && isReleaseWorthy(commitMessages) {
+			releaseNotes, err := createReleaseNotes(library, commitMessages, inputDirectory)
+			releaseVersion, err := calculateNextVersion(library)
+
+			//
+			if err := container.UpdateReleaseMetadata(flagImage, repoPath, inputDirectory, library.Id, releaseVersion); err != nil {
+				slog.Info(fmt.Sprintf("Received error running container: '%s'", err))
+
+			}
+			//TODO: make this configurable so we don't have to run per library
+			if flagSkipBuild {
+				if err := container.Build(flagImage, "repo-root", repoPath, "library-id", library.Id); err != nil {
+					slog.Info(fmt.Sprintf("Received error running container: '%s'", err))
+					//TODO: log in release PR
+				}
+			}
+
+			//TODO: add extra meta data what is this
+			prDescription += fmt.Sprintf("releasing lib: %s:%s\n", library.Id, library.NextVersion)
+			librariesToRelease[library.Id] = releaseVersion
+
+			createCommit(ctx, err, repo, releaseNotes)
+
+			//TODO: refactor, see if can wipe on every library instead of deleting
+			err = os.Remove(filepath.Join(inputDirectory, "release-notes.txt"))
+			if err != nil {
+				return nil, "", err
+			}
+		}
+	}
+	return librariesToRelease, prDescription, nil
+}
+
+func createCommit(ctx context.Context, err error, repo *gitrepo.Repo, releaseNotes string) {
+	_, err = gitrepo.AddAll(ctx, repo)
+	if err != nil {
+		fmt.Println("Error adding files:", err)
+		//TODO update PR description with this data and mark as not humanly resolvable
+	}
+	//TODO: what should the description be here?
+	if err := gitrepo.Commit(ctx, repo, releaseNotes); err != nil {
+		slog.Info(fmt.Sprintf("Received error trying to commit: '%s'", err))
+		//TODO update PR description with this data and mark as not humanly resolvable
+	}
+}
+
+// TODO: update with actual logic
+func createReleaseNotes(library *statepb.LibraryReleaseState, commitMessages []string, inputDirectory string) (string, error) {
+	releaseNotes := fmt.Sprintf("Library: %s\n", library.Id)
```

Comment:
I'd suggest the release notes file shouldn't include this - it should just be the changes.
(I realize all of this will change.)